bin/mastarm-deploy

```diff
@@ -28,86 +31,157 @@ commander
 
-const url = pkg.repository.url.replace('.git', '')
-const tag = `<${url}/commit/${commit}|${pkg.name}@${commit.slice(0, 6)}>`
-const config = loadConfig(process.cwd(), commander.config, commander.env)
-const get = util.makeGetFn([commander, config.settings])
+// each of these variables are also used in the logToMsTeams function and
+// these need to be defined after potentially decoding a sops-encoded file
+let cloudfront, config, env, minify, s3bucket, tag, url
 
-if (config.env.SLACK_WEBHOOK && config.env.SLACK_WEBHOOK.length > 0) {
-  logger.logToSlack({
-    channel: config.env.SLACK_CHANNEL || '#devops',
-    webhook: config.env.SLACK_WEBHOOK
-  })
-}
+async function deploy () {
+  // get information about the config directory being used
+  const repoUrl = await origin(commander.config)
+  let configCommit, configRemoteUrl
+  const configRepoRoot = repoInfo(commander.config).root
+  const configDir = path
+    .resolve(commander.config)
+    .replace(configRepoRoot, '')
 
-const files = util.parseEntries([...commander.args, ...(get('entries') || [])])
-util.assertEntriesExist(files)
-const sourceFiles = files.map(f => f[0])
-const outfiles = [...files.map(f => f[1]), ...files.map(f => `${f[1]}.map`)]
-
-const env = get('env') || 'development'
-const minify = get('minify')
-const buildOpts = {
-  config,
-  env,
-  files,
-  minify
-}
-const cloudfront = get('cloudfront')
-const s3bucket = get('s3bucket')
+  // check if the config directory being used is a configurations repo
+  if (repoUrl.endsWith('/configurations.git')) {
+    // run some extra checks to make sure configurations repo is up to date
+    // modified from https://stackoverflow.com/a/3278427/269834
+    configRemoteUrl = repoUrl.replace('.git', '')
+    const { stdout: local } = await execa('git', ['-C', configRepoRoot, 'rev-parse', '@'])
+    const { stdout: remote } = await execa('git', ['-C', configRepoRoot, 'rev-parse', '@{u}'])
+    const { stdout: base } = await execa('git', ['-C', configRepoRoot, 'merge-base', '@', '@{u}'])
+    let configurationsOk = false
+    if (local === remote) {
+      // Up-to-date
+      console.log('Configurations up-to-date')
+      configurationsOk = true
+    } else if (local === base) {
+      console.error('Configurations out of sync: Need to pull')
+    } else if (remote === base) {
+      console.error('Configurations out of sync: Need to push')
+    } else {
+      console.error('Configurations out of sync: Diverged')
+    }
+
+    // make sure there are no changes to the configurations
+    const { stdout: status } = await execa(
+      'git',
+      [
+        '-C',
+        configRepoRoot,
+        'status',
+        '-s'
+      ]
+    )
+    if (status === '') {
+      console.log('No changes to configurations repo')
+    } else {
+      console.error('Configurations out of sync: local changes exist that are not yet committed')
+      configurationsOk = false
+    }
+
+    if (!configurationsOk) process.exit(1)
```

Comment:
Can the above be refactored into a method called `checkConfigurationsRepo` (or something like that) that returns a boolean? That might make this script more readable.

Reply:
Actually, I was thinking about making this whole check it's own npm package. That way we could use it with the our server-side deployment scripts. I was thinking of calling it _git-repo-is-up-to-date_. Thoughts?

Reply:
Yea, I suppose it could be useful as an npm package. I think we should avoid spending a ton of time on the package meta stuff though.

Reply:
Ok, cool I'll make a new repo and npm package for that. It shouldn't take long. What do you mean by "package meta stuff"?

Reply:
Oh, just documentation, tests, etc. I think some documentation is called for though. I don't have a ton of experience publishing packages, so I don't really have a good sense for how much time goes into the whole process.